### PR TITLE
Behaviour Change for `_count` Operator - RN

### PR DESCRIPTION
**Pull request by JV0812**
## Purpose of this pull request

This pull request is to create a release note for the Behaviour Change for `_count` Operator.

## Select the type of change

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions, updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## Ticket (if applicable)

[Create release notes for count operator migration from Int to Long](https://sumologic.atlassian.net/browse/DOCS-389)

### CODE REVIEW

File: blog-service/2024-08-21-search.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+title: Behaviour Change for `_count` Operator (Search)
```
Comment (jpipkin1) [inline blog-service/2024-08-21-search.md:2]:
```suggestion
title: Behaviour Change for `count` Operator (Search)
```

**Review by jpipkin1 — APPROVED** [review]
Looks good, @JV0812. I had only one suggestion.

File: blog-service/2024-08-21-search.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - count
+  - search-operator
+  - log-search
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value after when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values.
```
Comment (manashar4) [inline blog-service/2024-08-21-search.md:14]: "...which could result in a negative count value ~after~ when exceeding the..."

File: blog-service/2024-08-21-search.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value after when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values. 
+
+This change will apply to all new queries. However, it may pose challenges for existing saved searches, scheduled searches, searches using the `count` operator as part of the scheduled view, and searches using the `count` operator as part of lookup tables. Below are some conditions to address these challenges:
+
+- For existing scheduled views, the query will use the `Int` data type, while new scheduled views will utilize the `Long` data type for the `count` operator.
```
Comment (manashar4) [inline blog-service/2024-08-21-search.md:18]: the scheduled view related callout should be added to the Scheduled Views page with a callout on the release note for users to expect newer scheduled views to support Long as the data type for count operator

File: blog-service/2024-08-21-search.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- For existing lookup tables, lookup schema matching has been relaxed to allow Int-Long type mismatch, in order to support this new behavioural change. If the schema type is `Int` and the field type is `Long`, there can be possible overflow, resulting in a negative count value after crossing the maximum integer value. If the schema type is `Long` and the field type is `Int` there will be no issues.
```
Comment (manashar4) [inline blog-service/2024-08-21-search.md:19]: the lookup related callout should be added to the lookup page with a callout on the release note for users to expect newer Lookup to only support Long as the data type for count operator

File: blog-service/2024-08-21-search.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+---
+title: Behaviour Change for `count` Operator (Search)
```
Comment (kimsauce) [inline blog-service/2024-08-21-search.md:2]:
```suggestion
title: Behavior Change for `count` Search Operator (Search)
```

File: blog-service/2024-08-21-search.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  - log-search
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values. 
+
+This change will apply to all new queries. Any existing queries as part of various content types - Log Search, Scheduled Search, and Saved Search will also seamlessly transition to show `Long` values in case of `Int` overflow.
```
Comment (kimsauce) [inline blog-service/2024-08-21-search.md:16]:
```suggestion
We're happy to announce an update to the behavior of our log search `count` operator. Previously, the `count` operator used the `Int` data type, which could lead to negative `count` values if the maximum integer value was exceeded. To address this, we have changed the data type of the `count` operator from `Int` to `Long`, allowing for the handling of larger numeric values without overflow.

This change will apply to all new queries. Existing queries in Log Search, Scheduled Search, and Saved Search will automatically transition to show `Long` values in cases where an `Int` overflow would have occurred.
```

**Comment by kimsauce** [comment]
Hi @manashar4 @JV0812 , will you be applying this update to the [`count` operator page](https://help.sumologic.com/docs/search/search-query-language/group-aggregate-operators/count-count-distinct-and-count-frequent/) as well?

File: blog-service/2024-08-21-search.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  - count
+  - search-operator
+  - log-search
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values.
```
Comment (manashar4) [inline blog-service/2024-08-21-search.md:14]: a change to the behaviour of the `count` operator to return `Long` values.  Previously, the `count` operator returned `Int` data type which would result in a negative value when exceeding the maximum integer value. With this change, we are changing  the data type of the `count` operator result from `Int` to `Long` data type.

File: blog-service/2024-08-21-search.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  - log-search
+hide_table_of_contents: true
+authors:
+  - url: https://help.sumologic.com/release-notes-service/rss.xml
+    image_url: /img/release-notes/rss-orange.png
+---
+
+We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values. 
+
+This change will apply to all new queries. Any existing queries as part of various content types - Log Search, [Scheduled Search](/docs/manage/scheduled-views/scheduled-views-best-practices/#upgrade-your-scheduled-views-to-avoid-negative-count-results), [Lookup Table](/docs/search/lookup-tables/manage-update-lookup-tables/#upgrade-the-lookup-table-to-avoid-negative-count-results), and Saved Search will also seamlessly transition to show `Long` values in case of `Int` overflow.
```
Comment (manashar4) [inline blog-service/2024-08-21-search.md:16]: hyperlink is for Scheduled View. so please add Scheduled View after scheduled Search and link this particular URL with scheduled view.

**Comment by JV0812** [comment]
@kimsauce Looks like we do not need to update this page as we have not explained anything related to the behaviour of the operator in this page.

File: docs/search/lookup-tables/manage-update-lookup-tables.md [hunk 9]
```diff
@@ -137,3 +137,18 @@ The status of an update can be one of the following:
...
+
+## FAQ
+
+### Upgrade the lookup table to avoid negative count results
+
+:::info
+Lookup schema matching is relaxed to allow Int-Long type mismatch to support the new behavioural change of the `count` operator.
+:::
+
+If the schema type is `Int` and the field type is `Long`, there can be possible overflow resulting in a negative count value after crossing the maximum integer value of `2147483647`. To migrate the schema type, create a new lookup table with `long` data type and run the below query:
```
Comment (manashar4) [inline docs/search/lookup-tables/manage-update-lookup-tables.md:150]: Replace "To migrate the schema type," with "As an Example To migrate the schema type,"

File: docs/search/lookup-tables/manage-update-lookup-tables.md [hunk 10]
```diff
@@ -137,3 +137,18 @@ The status of an update can be one of the following:
...
+
+```
+cat path://"/Library/Users/pyeole@sumologic.com/IntegerOverFlow-BeforeFF-count" | save path://"/Library/Users/pyeole@sumologic.com/IntegerOverFlowBeforeFFWithLong"
```
Comment (JV0812) [inline docs/search/lookup-tables/manage-update-lookup-tables.md:153]:
```suggestion
cat path://"/Library/Users/pyeole@sumologic.com/IntegerOverFlow-BeforeFF-count" | save path://"/Library/Users/pyeole@sumologic.com/IntegerOverFlowBeforeFFWithLong"
```

File: docs/search/lookup-tables/manage-update-lookup-tables.md (hunk 9 above)
Comment (JV0812) [reply docs/search/lookup-tables/manage-update-lookup-tables.md]: Updated this section.

**Pull request merged**